### PR TITLE
Add family planning referral

### PR DESCRIPTION
**Pull request by allan-allay**
Family planning referral form and functionality added as detailed in https://github.com/OpenSRP/opensrp-client-chw-family-planning/issues/18

### CODE REVIEW

File: opensrp-chw-core/src/main/java/org/smartregister/chw/core/presenter/CoreFamilyPlanningProfilePresenter.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        this.view = new WeakReference<>(view);
+        this.fpMemberObject = fpMemberObject;
+    }
+
+    @Override
+    public void createReferralEvent(AllSharedPreferences allSharedPreferences, String jsonString) throws Exception {
+        ((CoreFamilyPlanningMemberProfileContract.Interactor) interactor).createReferralEvent(allSharedPreferences, jsonString, fpMemberObject.getBaseEntityId());
+    }
+
+    @Override
```
Comment (paulinembabu) [inline opensrp-chw-core/src/main/java/org/smartregister/chw/core/presenter/CoreFamilyPlanningProfilePresenter.java]: This method could return a null, so i suggest annotate it with @Nullable

Comment (paulinembabu) [reply opensrp-chw-core/src/main/java/org/smartregister/chw/core/presenter/CoreFamilyPlanningProfilePresenter.java]: The rest seems fine to me

Comment (allan-allay) [reply opensrp-chw-core/src/main/java/org/smartregister/chw/core/presenter/CoreFamilyPlanningProfilePresenter.java]: Which method specifically?

Comment (paulinembabu) [reply opensrp-chw-core/src/main/java/org/smartregister/chw/core/presenter/CoreFamilyPlanningProfilePresenter.java]: getView

Comment (paulinembabu) [reply opensrp-chw-core/src/main/java/org/smartregister/chw/core/presenter/CoreFamilyPlanningProfilePresenter.java]: if (view != null) {
            return (CoreFamilyPlanningMemberProfileContract.View) view.get();
        } else {
            return null;
        }

**Pull request merged**